answer.js

```diff
@@ -1,17 +1,419 @@
+// });
+// // client or server error
+// app.use((err, req, res, next) => {
+// 	if (err.name === 'ValidationError') {
+// 		res.statusCode = 400;
+// 		return res.json({ error: err.message });
+// 	}
+// 	if (err.name === 'MongoError') {
+// 		return res.status(400).json({ error: err.message });
+// 	}
+// 	res.status(500).json({ err });
+// 	// console.log(err);
+// 	res.json({ error: err.message });
+// });
+
+// // listener
+// app.listen(PORT, () => {
+// 	console.log('Server started on', PORT);
+// });
+
+// Q11
+
+// let express = require('express');
+// let mongoose = require('mongoose');
+// let User = require('./models/users');
+// let app = express();
+// var PORT = process.env.PORT || 3000;
+
+// // connecting to mongodb database
+
+// mongoose.connect(
+// 	'mongodb://localhost/mongoose',
+// 	{
+// 		useNewUrlParser: true,
+// 		useUnifiedTopology: true,
+// 		useCreateIndex: true
+// 	},
+// 	(err) => {
+// 		console.log('Connected', err ? err : true);
+// 	}
+// );
+
+// //instantiate express app
+// app.use(express.json());
+// app.use(express.urlencoded({ extended: false }));
+
+// // Model.update
+
+// app.get('/:id', (req, res, next) => {
+// 	var objId = req.params.id;
+// 	User.update({ _id: objId }, (err, data) => {
+// 		if (err) return next(err);
+// 		res.status(200).json({ data });
+// 	});
+// });
+
+// // Model.updateOne
+
+// app.get('/:id', (req, res, next) => {
+// 	var objId = req.params.id;
+// 	User.updateOne({ _id: objId }, (err, data) => {
+// 		if (err) return next(err);
+// 		res.status(200).json({ data });
+// 	});
+// });
+
+// // model.findByIdAndUpdate(id)
+
+// app.get('/:id', (req, res, next) => {
+// 	var objId = req.params.id;
+// 	User.findByIdAndUpdate({ objId }, (err, data) => {
+// 		if (err) return next(err);
+// 		res.status(200).json({ data });
+// 	});
+// });
+// //error handler middlewares
+// // 404
+// app.use((req, res, next) => {
+// 	res.status(404).send('<h1> Page not found </h1>');
+// });
+// // client or server error
+// app.use((err, req, res, next) => {
+// 	if (err.name === 'ValidationError') {
+// 		res.statusCode = 400;
+// 		return res.json({ error: err.message });
+// 	}
+// 	if (err.name === 'MongoError') {
+// 		return res.status(400).json({ error: err.message });
+// 	}
+// 	res.status(500).json({ err });
+// 	// console.log(err);
+// 	res.json({ error: err.message });
+// });
+
+// // listener
+// app.listen(PORT, () => {
+// 	console.log('Server started on', PORT);
+// });
+
+// Q12
+
+let express = require('express');
+let mongoose = require('mongoose');
+let User = require('./models/users');
+let app = express();
+var PORT = process.env.PORT || 3000;
+
+// connecting to mongodb database
+
+mongoose.connect(
+	'mongodb://localhost/mongoose',
+	{
+		useNewUrlParser: true,
+		useUnifiedTopology: true,
+		useCreateIndex: true
+	},
+	(err) => {
+		console.log('Connected', err ? err : true);
+	}
+);
+
+//instantiate express app
```

Comment:
Same as above.
You have not instantiated by mounting express on app variable like `var app = express()`